### PR TITLE
flash : stm32l4x don't write 0xff

### PR DESCRIPTION
**Pull request by jamike**
for mcuboot on l4, avoiding writting 0xff solves trailer write access during swap

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Any reason not to include this in PR #388?

Also similar comment about subject of commit message provide additional details about the issue & fix.

**Comment by lpereira** [comment]
You might want to rebase this, so it only includes the last patch. You can force-push to your repository, this PR will be updated.

**Comment by jamike** [comment]
This commit is required is required only for MCU boot loader. That 's why I pushed it separetly.

File: drivers/flash/flash_stm32l4x.c [hunk 1]
```diff
@@ -142,9 +142,13 @@ int flash_stm32_write_range(unsigned int offset, const void *data,
 	int i, rc = 0;
 
 	for (i = 0; i < len; i += 8, offset += 8) {
-		rc = write_dword(offset, ((const u64_t *) data)[i], p);
-		if (rc < 0) {
-			return rc;
+		if (((const u64_t *) data)[i>>3] != -1) {
+			rc = write_dword(offset,
+					((const u64_t *)data)[i>>3]
+					, p);
```
Comment (erwango) [inline drivers/flash/flash_stm32l4x.c]: can you move coma at end of second line?

**Review by erwango — CHANGES_REQUESTED** [review]
minor comment, otherwise LGTM

**Comment by galak** [comment]
now that the other flash changes are merged, can you rebase/update this.

**Comment by jamike** [comment]
Following some changes in mcu bootloader . this patch seems not needed . I continue testing without. 
And if confirmed , i will close it.

**Comment by galak** [comment]
any updates if still needed?

**Comment by jamike** [comment]
no

**Comment by jamike** [comment]
After the changes in mcuboot , the PR is not needed any more.

**Pull request closed without merging**